alfresco-search/src/test/java/org/alfresco/solr/query/AFTSRangeQueryTest.java

```diff
@@ -0,0 +1,136 @@
+        String stringName2 = "test2";
+        String rating2 = "test";
+        String creator2 = "Luigi";
+
+        NodeRef nodeRef2 = new NodeRef(new StoreRef("workspace", "SpacesStore"), guid);
+        QName name2 = QName.createQName(NamespaceService.CONTENT_MODEL_1_0_URI, stringName2);
+        HashMap<QName, PropertyValue> properties2 = new HashMap<QName, PropertyValue>();
+        properties2.put(ContentModel.PROP_NAME, new StringPropertyValue(stringName2));
+        properties2.put(ContentModel.PROP_RATING_SCHEME, new StringPropertyValue(rating2));
+        properties2.put(ContentModel.PROP_CREATOR, new StringPropertyValue(creator2));
+        addNode(core, dataModel, 1, 3, 1, ContentModel.TYPE_CONTENT, aspects, properties2, null, "elia",
+                null, new NodeRef[]{rootNodeRef}, new String[]{"/"
+                        + name2.toString()}, nodeRef2, true);
+        System.out.println("create node test2 " + guid);
+
+        guid = createGUID();
+        String stringName3 = "test3";
+        String rating3 = "firstString";
+        String creator3 = "Wario";
+
+        NodeRef nodeRef3 = new NodeRef(new StoreRef("workspace", "SpacesStore"), guid);
+        QName name3 = QName.createQName(NamespaceService.CONTENT_MODEL_1_0_URI, stringName3);
+        HashMap<QName, PropertyValue> properties3 = new HashMap<QName, PropertyValue>();
+        properties3.put(ContentModel.PROP_NAME, new StringPropertyValue(stringName3));
+        properties3.put(ContentModel.PROP_RATING_SCHEME, new StringPropertyValue(rating3));
+        properties3.put(ContentModel.PROP_CREATOR, new StringPropertyValue(creator3));
+        addNode(core, dataModel, 1, 4, 1, ContentModel.TYPE_CONTENT, aspects, properties3, null, "elia",
+                null, new NodeRef[]{rootNodeRef}, new String[]{"/"
+                        + name3.toString()}, nodeRef3, true);
+        System.out.println("create node test3 " + guid);
+    }
+
+    @Test
+    public void testRangeQueriesNonTokenised() throws Exception
+    {
+        // test on cm:ratingScheme that is non tokenised.
+        assertQ(areq(params("rows", "20", "qt", "/afts", "q", "cm:ratingScheme:[n TO *]"), null), "*[count(//doc)=2]");
+        assertQ(areq(params("rows", "20", "qt", "/afts", "q", "cm:ratingScheme:[* TO n]"), null), "*[count(//doc)=1]");
+        assertQ(areq(params("rows", "20", "qt", "/afts", "q", "cm:ratingScheme:[first TO *]"), null), "*[count(//doc)=3]");
+        assertQ(areq(params("rows", "20", "qt", "/afts", "q", "cm:ratingScheme:[firstt TO *]"), null), "*[count(//doc)=2]");
+        assertQ(areq(params("rows", "20", "qt", "/afts", "q", "cm:ratingScheme:[* TO someTest]"), null), "*[count(//doc)=2]");
+        assertQ(areq(params("rows", "20", "qt", "/afts", "q", "cm:ratingScheme:[kk TO someTest]"), null), "*[count(//doc)=1]");
+    }
+
+    @Test
+    public void testRangeQueriesTokenised() throws Exception
+    {
+        // test on cm:ratingScheme that is non tokenised.
```

Comment:
"test on cm:ratingScheme that is tokenised"

Reply:
That's right. Thanks